palindrom-client.html

```diff
@@ -284,67 +290,84 @@
+                                        addedCount: 1,
+                                        object: parent,
+                                        type: 'splice'
+                                    });
                                     break;
                                 case 'remove':
-                                    templateDomBind.notifySplices(parentsPolymerPath,
-                                        [{
-                                            index: parseInt(name, 10),
-                                            removed: (result && [result.removed]) || [],
-                                            addedCount: 0,
-                                            object: parent,
-                                            type: 'splice'
-                                        }]
-                                    );
+                                    splices[parentsPolymerPath].push({
+                                        index: parseInt(name, 10),
+                                        removed: (result && [result.removed]) || [],
+                                        addedCount: 0,
+                                        object: parent,
+                                        type: 'splice'
+                                    });
                                     break;
                             }
                         } else {
                             switch (operation.op) {
-
                                 case 'add':
                                 case 'replace':
                                     if (typeof operation.value === 'object') {
-                                        templateDomBind.notifyPath(polymerPath);
+                                        notifications[polymerPath] = [];
                                     } else {
-                                        templateDomBind.notifyPath(polymerPath, operation.value, false);
+                                        notifications[polymerPath] = [operation.value, false];
                                     }
                                     break;
                                 case 'remove':
-                                    //console.warn('Polymer does not support unsetting properties https://github.com/Polymer/polymer/issues/2565');
-                                    //console.warn('remove is translated to JSON incompatible set to `undefined`, as there is no remove method in Polymer.')
-                                    templateDomBind.notifyPath(polymerPath, null, false);
-                                    templateDomBind.notifyPath(polymerPath, undefined, false);
+                                    notifications[polymerPath] = [null, false];
                                     break;
                             }
                         }
                     }
                 }
+                /**
+                *  STEP 2: APPLY THE UPDATES
```

Comment:
Why was it important to make the patches unique by path? Were there duplicates sent by the server?

If yes, then I think that should be corrected in the server implementation.

Reply:
The UNIQ by path part does not fix an existing problem. It's just a coincidence bonus side effect due to using an object to store updates. I can't see any reason to apply two updates to the same path in a synchronous for loop. The first update will never be rendered anyways. And it applies only to updating objects (as opposed to arrays).

Also, regarding splices (updaing arrays), there is no UNIQ by path, there is clustering by path. This makes notify splices to be called like this:
```js
notifySplices([splice1]);

Reply:
> The UNIQ by path part does not fix an existing problem. It's just a coincidence bonus side effect due to using an object to store updates. I can't see any reason to apply two updates to the same path in a synchronous for loop. The first update will never be rendered anyways. And it applies only to updating objects (as opposed to arrays).

Reply:
This code is gone now.

Reply:
```js

Reply:
But we'll only benefit from it up from patches that are up to 100 operations. Do you think it's worth it?